docs/3-building-blocks-of-oop-part-2/hometask/src/Item.ts

```diff
@@ -1,11 +1,79 @@
 import { Comparable } from './Comparable';
 
 let id = 0;
+let counter = 0;
 
 export abstract class Item implements Comparable<Item> {
-    public compareTo(other: Item): number {
-        // your code goes here
+  private id: number;
+  private value: number;
+  private weight: number;
+  private name: string;
+
+  constructor(name: string, value: number, weight: number) {
+    this.id = id;
+    this.name = name;
+    this.value = value;
+    this.weight = weight;
+    id += 1;
+    counter += 1;
+  }
+  public use(): void {}
+
+  public compareTo(other: Item): number {
+    if (this.getValue() > other.getValue()) {
+      return 1;
+    } else if (this.getValue() < other.getValue()) {
+      return -1;
+    } else {
+      const name = this.getName().toLowerCase();
```

Comment:
We always should use `localCompare` to compare strings in JS - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare